erts/emulator/beam/dist.c

```diff
@@ -6205,15 +6362,62 @@ erts_monitor_nodes(Process *c_p, Eterm on, Eterm olist)
     if (on != am_true && on != am_false)
 	return THE_NON_VALUE;
 
-    if (is_not_nil(opts_list)) {
+    if (is_map(options)) {
```

Comment:
Could we move some of this code to net_kernel:monitor_nodes? That is something like this:

```erlang
monitor_nodes(Flag, Opts) when is_list(Opts) ->
    case catch process_flag({monitor_nodes, maps:from_list(Opts)#{ info_type => list }}, Flag) of
	N when is_integer(N) -> ok;
	_ -> mk_monitor_nodes_error(Flag, Opts)
    end;
monitor_nodes(Flag, Opts) when is_map(Opts) ->
    case catch process_flag({monitor_nodes, Opts#{ info_type => map }}, Flag) of
	N when is_integer(N) -> ok;
	_ -> mk_monitor_nodes_error(Flag, Opts)
    end;
```

Reply:
Implemented